### PR TITLE
show deprecation for additionalAssets hook

### PR DESCRIPTION
**Pull request by chenxsan**
**What kind of change does this PR introduce?**

Show deprecation when users use `additionalAssets` hook:

> (node:29478) [DEP_WEBPACK_COMPILATION_ADDITIONAL_ASSETS] DeprecationWarning: additionalAssets is deprecated (use Compilation.hook.processAssets instead and use one of Compilation.PROCESS_ASSETS_STAGE_* as stage option)

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

I'll add a notice to document for the hook.

### CODE REVIEW

File: lib/Compilation.js [hunk 1]
```diff
@@ -517,7 +517,8 @@ BREAKING CHANGE: Asset processing hooks in Compilation has been merged into a si
 			additionalAssets: createProcessAssetsHook(
 				"additionalAssets",
 				Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL,
-				() => []
+				() => [],
+				"dep_webpack_compilation_additional_assets".toUpperCase()
```
Comment (evilebottnawi) [inline lib/Compilation.js:521]: It is soft deprecation without warnings

Comment (chenxsan) [reply lib/Compilation.js:521]: Only hooks will be removed in webpack 6 show those warnings?

Comment (evilebottnawi) [reply lib/Compilation.js:521]: in webpack@6 we will emit deprecation for this

Comment (chenxsan) [reply lib/Compilation.js:521]: Got it.

**Pull request closed without merging**